app/assets/javascripts/darkswarm/directives/shop_variant.js.coffee

```diff
@@ -6,4 +6,4 @@ Darkswarm.directive "shopVariant", ->
     variant: '='
   controller: ($scope, Cart) ->
     $scope.$watchGroup ['variant.line_item.quantity', 'variant.line_item.max_quantity'], ->
-      Cart.adjust($scope.variant.line_item)
+      Cart.adjust($scope.variant.line_item) if $scope.variant.line_item.quantity > 0
```

Comment:
I think we need to send quantity = 0 to remove the line_item :warning: 
have you tested that adding a product and then reducing quantity to zero again will work?

Reply:
yeah, right. No, I haven't gone into much detail while testing. I'll do that next.